contracts/whitelist/src/contract.rs

```diff
@@ -157,35 +178,32 @@ pub fn execute_update_end_time(
         .add_attribute("sender", info.sender))
 }
 
-pub fn execute_update_members(
+pub fn execute_add_members(
     deps: DepsMut,
     _env: Env,
     info: MessageInfo,
-    msg: UpdateMembersMsg,
+    msg: AddMembersMsg,
 ) -> Result<Response, ContractError> {
     let mut config = CONFIG.load(deps.storage)?;
     if info.sender != config.admin {
         return Err(ContractError::Unauthorized {});
     }
 
-    for add in msg.add.into_iter() {
+    for add in msg.to_add.into_iter() {
         if config.num_members >= MAX_MEMBERS {
             return Err(ContractError::MembersExceeded {
                 expected: MAX_MEMBERS,
                 actual: config.num_members,
             });
         }
         let addr = deps.api.addr_validate(&add)?;
+        if WHITELIST.has(deps.storage, addr.clone()) {
+            return Err(ContractError::DuplicateMember(addr.to_string()));
+        }
```

Comment:
we could also do
```
if !WHITELIST.has(deps.storage, addr.clone()) {
        WHITELIST.save(deps.storage, addr, &true)?;
        config.num_members += 1;
}
```
but maybe better to be explicit about the issue

Reply:
This is how I had it originally, but decided to be explicit in case they made any errors and uploading a lot of duplicates.